pvlib/data/gold/sdm.py

```diff
@@ -0,0 +1,847 @@
+"""
+The ``sdm`` module contains functions for generating gold current-voltage
+(I-V) data for the five-parameter single-diode model (SDM) and gauging
+functions that compute I-V values in terms of accuracy and computation speed.
+
+Run this module from the command line to generate a gold dataset in two
+formats, pvlib/data/gold/sdm.json (lossles, machine-interoperable) and
+pvlib/data/gold/sdm.csv (human readable and lossy) and then guage functions
+against it that are compatible with the pvlib API.
+"""
+
+import copy
+import datetime
+import json
+import os
+import platform
+import sys
+from timeit import default_timer as timer
+
+import numpy
+
+import pvlib
+
+
+# Useful constants for photovoltaic modeling that are versioned. Several
+# constants are derived from scipy.constants v1.1.0, which are from
+# standardized CODATA. See https://physics.nist.gov/cuu/Constants/. We do not
+# import scipy because it is not required in the base pvlib installation. Units
+# appear in the variable suffix to eliminate ambiguity.
+
+# elementary charge, from scipy.constants.value('elementary charge')
+elementary_charge_C = 1.6021766208e-19
+
+# Boltzmann constant, from scipy.constants.value('Boltzmann constant')
+boltzmann_J_per_K = 1.38064852e-23
+
+# Define standard test condition (STC) temperature in degrees Celsius
+T_stc_degC = 25.
+
+# Define standard test condition (STC) temperature in Kelvin
+T_stc_K = 273.15 + T_stc_degC
+
+
+def sum_current(resistance_shunt, resistance_series, nNsVth, current, voltage,
```

Comment:
Minor : would be nice to have the positional arguments in the same order as in `pvsystem.singlediode`

Reply:
Belay that.  The arguments are in the same order as for `v_from_i` and `i_from_v`